src/allmydata/frontends/magic_folder.py

```diff
@@ -1053,6 +1064,22 @@ def _process_deque(self):
 
                 self._call_hook(proc, 'processed')
 
+    def _trim_process_history(self):
+        """
+        Remove any history items which are 'too old' (which is currently
+        defined as more than 30s).
+        """
+        now = time.time()
```

Comment:
This needs a parameterized clock for easier testing.  Fortunately `self._clock` is already a thing.